File: csharp/src/Behavioral/Interpreter/DvdTitleExpression.cs
//DvdTitleExpression.java - Three Of Four Terminal Expressions

namespace Behavioral.Interpreter {

using System;
using System.Collections.Generic;


public class DvdTitleExpression : DvdAbstractExpression {  
   override public string interpret(DvdInterpreterContext 
                           dvdInterpreterContext) {
       List<string> titles = dvdInterpreterContext.getAllTitles();
       List<string> titleBuffer = new List<string>();
       bool first = true;
       foreach( string title in titles )
			{
           if (!first)
               titleBuffer.Add(", ");
           else
               first = false;

           titleBuffer.Add(title );
       }
       return titleBuffer.ToString();
   }
}

}

File: scala/src/main/scala/Behavioral/Interpreter/DvdTitleExpression.scala
package Behavioral.Interpreter;


//DvdTitleExpression - Three Of Four Terminal Expressions

import java.util.ArrayList;
import java.util.ListIterator;

public class DvdTitleExpression extends DvdAbstractExpression {  
   public String interpret(DvdInterpreterContext 
                           dvdInterpreterContext) {
       ArrayList titles = dvdInterpreterContext.getAllTitles();
       ListIterator titlesIterator = titles.listIterator();
       StringBuffer titleBuffer = new StringBuffer("");
       boolean first = true;
       while (titlesIterator.hasNext()) {
           if (!first) {
               titleBuffer.append(", ");
           } else {
               first = false;
           }    
           titleBuffer.append((String)titlesIterator.next());
       }
       return titleBuffer.toString();
   }
}